test/functional/p2p_mutated_blocks.py

```diff
@@ -111,6 +118,23 @@ def self_transfer_requested():
                 attacker.send_message(msg_block(block_missing_prev))
         attacker.wait_for_disconnect(timeout=5)
 
+        # Test that we log when receiving a previously unknown, valid-PoW
+        # mutated block.
+        # Mine a block and mutate it
+        tx = self.wallet.create_self_transfer()["tx"]
+        block = create_block(tmpl=self.nodes[0].getblocktemplate(NORMAL_GBT_REQUEST_PARAMS), txlist=[tx])
+        add_witness_commitment(block)
+        block.solve()
+        mutated_block = copy.deepcopy(block)
+        mutated_block.vtx[0].wit = CTxWitness()
+
+        # Send the mutated block and check that the rejection was logged.
+        attacker = self.nodes[0].add_p2p_connection(P2PInterface())
+        with self.nodes[0].assert_debug_log(
+          expected_msgs=["bad-witness-nonce-size, CheckWitnessMalleation", LOGMESSAGE_REJECTED_MUTATED_BLOCK]):
+            attacker.send_message(msg_block(mutated_block))
+        attacker.wait_for_disconnect(timeout=5)
+
```

Comment:
Logic seems wrong, and here's a test which catches it. We'll make an unconditional log for every mutated block, over and over. To mitigate this we would need to accept the header of the mutated block and use that to rate-limit the log.
```suggestion

        # But won't log a second time
        attacker = self.nodes[0].add_p2p_connection(P2PInterface())
        with self.nodes[0].assert_debug_log(
          expected_msgs=["Received mutated block from peer"], # have it wait for this to make sure unexpected has time
          unexpected_msgs=["bad-witness-nonce-size, CheckWitnessMalleation", LOGMESSAGE_REJECTED_MUTATED_BLOCK]):
            attacker.send_message(msg_block(mutated_block))
        attacker.wait_for_disconnect(timeout=5)

```

Reply:
This was mentioned in https://github.com/bitcoin/bitcoin/pull/29412#issuecomment-1972837712 too. The current implementation in this PR is the same behavior as before #29412. The logging is somewhat rate-limited by needing to open a new connection each time.

I agree that re-introducing this isn't ideal.